test/python/quantum_info/operators/symplectic/test_pauli_list.py

```diff
@@ -2071,6 +2072,51 @@ def test_evolve_clifford_qargs(self, phase):
         self.assertListEqual(value, value_h)
         self.assertListEqual(value_inv, value_s)
 
+    def test_qubit_wise_noncommutation_graph(self):
+        """Test qubit-wise noncommutation graph"""
+
+        def qubitwise_commutes(left: Pauli, right: Pauli) -> bool:
+            return len(left) == len(right) and all(a.commutes(b) for a, b in zip(left, right))
+
+        input_labels = ["IY", "ZX", "XZ", "YI", "YX", "YY", "YZ", "ZI", "ZX", "ZY", "iZZ", "II"]
+        np.random.shuffle(input_labels)
+        pauli_list = PauliList(input_labels)
+        graph = pauli_list.noncommutation_graph(qubit_wise=True)
+
+        expected = rx.PyGraph()
+        expected.add_nodes_from(range(len(input_labels)))
+        edges = [
+            (ia, ib)
+            for (ia, a), (ib, b) in itertools.combinations(enumerate(input_labels), 2)
+            if not qubitwise_commutes(Pauli(a), Pauli(b))
+        ]
+        expected.add_edges_from_no_data(edges)
+
+        self.assertTrue(rx.is_isomorphic(graph, expected))
+
+    def test_noncommutation_graph(self):
+        """Test noncommutation graph"""
+
+        def commutes(left: Pauli, right: Pauli) -> bool:
+            return len(left) == len(right) and left.commutes(right)
+
+        input_labels = ["IY", "ZX", "XZ", "YI", "YX", "YY", "YZ", "ZI", "ZX", "ZY", "iZZ", "II"]
+        np.random.shuffle(input_labels)
+        pauli_list = PauliList(input_labels)
+        #  if qubit_wise=True, equivalent to test_qubit_wise_noncommutation_graph
+        graph = pauli_list.noncommutation_graph(qubit_wise=False)
+
+        expected = rx.PyGraph()
+        expected.add_nodes_from(range(len(input_labels)))
+        edges = [
+            (ia, ib)
+            for (ia, a), (ib, b) in itertools.combinations(enumerate(input_labels), 2)
+            if not commutes(Pauli(a), Pauli(b))
+        ]
+        expected.add_edges_from_no_data(edges)
+
+        self.assertTrue(rx.is_isomorphic(graph, expected))
```

Comment:
This is done in the same style as the `group_commuting` tests below. In the `SparsePauliOp` test cases these are parameterized tests. We could do that here, too, if we want to avoid code duplication, but I was not sure if there was another reason for not having the ones inside `PauliList` parameterized,

I can update this if desired

Reply:
I think the `SparsePauliOp` tests got rewritten somewhat after they'd originally gone in, which is probably where the discrepancy arose.  I don't really mind either way if you want to make the tests match between files - I think either is fine, so whatever you think is best/easier.